src/Crate/PDO/Http/ServerPool.php

```diff
@@ -0,0 +1,227 @@
+<?php
+/**
+ * Licensed to CRATE Technology GmbH("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+namespace Crate\PDO\Http;
+
+use Crate\PDO\Exception\RuntimeException;
+use Crate\PDO\PDO;
+use Crate\PDO\PDOInterface;
+use Crate\Stdlib\Collection;
+use Crate\Stdlib\CollectionInterface;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\BadResponseException;
+use GuzzleHttp\Exception\ConnectException;
+use GuzzleHttp\RequestOptions;
+
+final class ServerPool implements ServerInterface
+{
+    private const DEFAULT_SERVER = 'localhost:4200';
+
+    /**
+     * @var string
+     */
+    private $protocol = 'http';
+
+    /**
+     * @var array
+     */
+    private $httpOptions = [];
+
+    /**
+     * @var string[]
+     */
+    private $availableServers = [];
+
+    /**
+     * @var Client
+     */
+    private $httpClient;
+
+    /**
+     * Client constructor.
+     *
+     * @param array $servers
+     */
+    public function __construct(array $servers)
+    {
+        if (\count($servers) === 0) {
+            $servers = [self::DEFAULT_SERVER];
+        }
+
+        // micro optimization so we don't always hit the same server first
+        shuffle($servers);
+
+        foreach ($servers as $server) {
+            $this->availableServers[] = $server;
+        }
+
+        $this->httpClient = new Client();
```

Comment:
Worth making this an injectable: the test suite already uses reflection to workaround this limitation, and I can already see folks willing to replace the `Client` due to edge cases. It's a simple addition with no evident downsides